packages/uniforms/__tests__/BaseForm.tsx

```diff
@@ -1,387 +1,76 @@
-        readOnly: false,
-        showInlineError: true,
-      });
-    });
-
-    it('updates schema bridge', () => {
-      const schema2 = { ...(schema as Omit<Bridge, never>), getType: () => {} };
-
-      wrapper.setProps({ schema: schema2 });
-
-      const context = wrapper.instance().getContext();
-
-      expect(context).toHaveProperty('schema', schema2);
-    });
-
-    it('ignores changes made on first render', () => {
-      function Field() {
-        const [props] = useField('name', {});
-        props.onChange(123);
-        return null;
-      }
-
-      const wrapper = mount<BaseForm<any>>(
-        <BaseForm onChange={onChange} schema={schema}>
-          <Field />
+      const { container } = render(
+        <BaseForm schema={schema}>
+          <div />
+          <div />
+          <div />
         </BaseForm>,
       );
 
-      const context = wrapper.instance().getContext();
-      expect(context).toHaveProperty('changed', false);
-      expect(context).toHaveProperty('changedMap', {});
-
-      expect(onChange).toHaveBeenCalledTimes(1);
-      expect(onChange).toHaveBeenCalledWith('name', 123);
-    });
-  });
-
-  describe('when changed', () => {
-    type Form = BaseForm<any>;
-    let wrapper: ReactWrapper<Form['props'], Form['state'], Form>;
-
-    beforeEach(() => {
-      wrapper = mount(
-        <BaseForm
-          model={model}
-          schema={schema}
-          onChange={onChange}
-          onSubmit={onSubmit}
-        />,
-      );
-    });
-
-    it('updates `changed` and `changedMap`', () => {
-      const context1 = wrapper.instance().getContext();
-      expect(context1).toHaveProperty('changed', false);
-      expect(context1).toHaveProperty('changedMap', {});
-
-      wrapper.instance().getContext().onChange('$', [1, 2]);
-
-      const context2 = wrapper.instance().getContext();
-      expect(context2).toHaveProperty('changed', true);
-      expect(context2).toHaveProperty('changedMap.$');
-      expect(context2.changedMap.$).toBeTruthy();
-      expect(context2).toHaveProperty('changedMap.$.1');
-      expect(context2.changedMap.$?.[1]).toBeTruthy();
-
-      wrapper.instance().getContext().onChange('$', [1]);
-
-      const context3 = wrapper.instance().getContext();
-      expect(context3).toHaveProperty('changed', true);
-      expect(context3).toHaveProperty('changedMap.$');
-      expect(context3.changedMap.$).toBeTruthy();
-      expect(context3).toHaveProperty('changedMap.$.1');
-      expect(context3.changedMap.$?.[1]).toBeTruthy();
-    });
-
-    it('autosaves correctly (`autosave` = true)', async () => {
-      wrapper.setProps({ autosave: true });
-      wrapper.instance().getContext().onChange('a', 1);
-      await new Promise(resolve => setTimeout(resolve));
-      const context = wrapper.instance().getContext();
-      expect(onSubmit).toHaveBeenCalledTimes(1);
-      expect(context.submitted).toBe(true);
-      expect(onSubmit).toHaveBeenLastCalledWith(model);
-    });
```

Comment:
We could rewrite it with `fireEvent` and check how many times did `onSubmit` was called.